clcache.py

```diff
@@ -1165,7 +1253,11 @@ def processCompileRequest(cache, compiler, args):
 
 def processDirect(cache, outputFile, compiler, cmdLine, sourceFile):
     manifestHash = ObjectCache.getManifestHash(compiler, cmdLine, sourceFile)
-    with cache.lock:
+    postProcessing = None
+
+    try:
```

Comment:
Sorry some temporary change. I revert it. I wanted to try if it is possible to fall back to just invoke the compiler in case we hit a timeout waiting for the lockfile instead of aborting the build.
Before adding the random pollDelay I observed some race conditions, where 12 clcache process were fighting for the lockfile and no one made progress.